azure/test.yml

```diff
@@ -24,33 +25,38 @@ jobs:
           vmImage: macOS-10.14
         Windows:
           vmImage: windows-2019
-  ${{ if ne('true', parameters.cross) }}:
-    variables:
+  variables:
+    ${{ if ne('true', parameters.cross) }}:
       vmImage: ubuntu-16.04
+    options: ''
   pool:
     vmImage: $(vmImage)
   steps:
   - template: _setup.yml
     parameters:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
-  - ${{ if ne('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}"
-      displayName: Run tests
-      env:
-        ${{ insert }}: ${{ parameters.envs }}
-  - ${{ if eq('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}" -- --test-threads=1
-      displayName: Run tests (single-threaded)
+  # Cargo options go here
+  - script: echo '##vso[task.setvariable variable=options]$(options) --all'
+  - ${{ if eq(true, parameters.features)}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --features "${{ parameters.features }}"'
+  # Add any additional conditional libtest options to this check
+  - ${{ if or(eq(true, parameters.single_threaded), eq(true, parameters.test_ignored))}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --'
+  # Options passed to libtest go here
+  - ${{ if eq(true, parameters.single_threaded) }}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --test-threads=1'
+  - ${{ if eq(true, parameters.test_ignored)}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --ignored'
```

Comment:
This is not quite right. `--ignored` runs _only_ ignored tests. So we actually need to invoke `cargo test` a second time if `test_ignored` is on with the same `options` but _also_ `--ignored`.

Reply:
If that is the case (and looking at `cargo test -- -h` that seems correct) than the documentation for "Ignored Tests" [here](https://crate-ci.github.io/azure-pipelines/configuration.html) is erroneous. There is a libtest option (--include-ignored) which seems to do what the documentation suggests is supposed to happen. What is the intended functionality of `test_ignored` ?

Reply:
Why do you believe those docs are erroneous? Following https://github.com/crate-ci/azure-pipelines/pull/49, when you pass `test_ignored: true` then tests with `#[ignored]` are _also_ run as the docs say (that is, we run both `cargo test` _and_ `cargo test -- --ignored`).

Reply:
So `test_ignored` should run both ignored and non-ignored tests?

Reply:
Correct!